pycalphad/plot/eqplot.py

```diff
@@ -55,62 +66,116 @@ def eqplot(eq, ax=None, x=None, y=None, z=None, phases=None, **kwargs):
+
+    # For both two and three phase, cast the tuple of indices to an array and flatten
+    # If we found two phase regions:
+    if len(np.array(two_phase_idx).flatten()) > 0:
+        found_two_phase = eq.Phase.values[two_phase_idx][..., :2]
+        # get tieline endpoint compositions
+        two_phase_x = eq.X.sel(component=x.species).values[two_phase_idx][..., :2]
+        # handle special case for potential
+        if isinstance(y, v.Composition):
+            two_phase_y = eq.X.sel(component=y.species).values[two_phase_idx][..., :2]
+        else:
+            # it's a StateVariable. This must be True
+            two_phase_y = np.take(eq[str(y)].values, two_phase_idx[list(str(i) for i in conds.keys()).index(str(y))])
+            # because the above gave us a shape of (n,) instead of (n,2) we are going to create it ourselves
+            two_phase_y = np.array([two_phase_y, two_phase_y]).swapaxes(0, 1)
+
+        # construct tielines
+        tielines = np.array([np.concatenate((two_phase_x[..., 0][..., np.newaxis], two_phase_y[..., 0][..., np.newaxis]), axis=-1),
+                     np.concatenate((two_phase_x[..., 1][..., np.newaxis], two_phase_y[..., 1][..., np.newaxis]), axis=-1)])
+        tielines = np.rollaxis(tielines,1)
+        lc = mc.LineCollection(tielines, zorder=1, colors=[0,1,0,1], linewidths=[0.5, 0.5])
+        # plot two phase points and tielines
+        two_phase_plotcolors = np.array(list(map(lambda x: [colorlist[x[0]], colorlist[x[1]]], found_two_phase)), dtype='U') # from pycalphad
+        ax.scatter(two_phase_x[..., 0], two_phase_y[..., 0], s=3, c=two_phase_plotcolors[:,0], edgecolors='None', zorder=2, **kwargs)
+        ax.scatter(two_phase_x[..., 1], two_phase_y[..., 1], s=3, c=two_phase_plotcolors[:,1], edgecolors='None', zorder=2, **kwargs)
+        ax.add_collection(lc)
+
+    # If we found three phase regions:
+    if len(np.array(three_phase_idx).flatten()) > 0:
+        found_three_phase = eq.Phase.values[three_phase_idx][..., :3]
+        # get tieline endpoints
+        three_phase_x = eq.X.sel(component=x.species).values[three_phase_idx][..., :3]
+        three_phase_y = eq.X.sel(component=y.species).values[three_phase_idx][..., :3]
+        # three phase tielines
+        three_phase_tielines = np.array([np.concatenate((three_phase_x[..., 0][..., np.newaxis], three_phase_y[..., 0][..., np.newaxis]), axis=-1),
+                                     np.concatenate((three_phase_x[..., 1][..., np.newaxis], three_phase_y[..., 1][..., np.newaxis]), axis=-1),
+                                     np.concatenate((three_phase_x[..., 2][..., np.newaxis], three_phase_y[..., 2][..., np.newaxis]), axis=-1)])
+        three_phase_tielines = np.rollaxis(three_phase_tielines,1)
+        three_lc = mc.LineCollection(three_phase_tielines, zorder=1, colors=[1,0,0,1], linewidths=[0.5, 0.5])
+        # plot three phase points and tielines
+        # TODO: the colors seem to be ordered wrong on the plots
```

Comment:
The coloring of the markers seemed wrong on the tie triangles I was looking at. The three marker colors were usually correct, but they were on the wrong vertex of the triangle.